lib/mongo/collection.rb

```diff
@@ -239,6 +244,12 @@ def create(opts = {})
       options = Hash[self.options.reject do |key, value|
         %w(read read_preference read_concern).include?(key.to_s)
       end]
+      # Converting Ruby spelled time series options to server style.
+      [[:time_series,  :timeseries], [:expire_after, :expireAfterSeconds]].each do |(ruby_key, server_key)|
+        if options.has_key?(ruby_key)
+          options[server_key] = options.delete(ruby_key)
```

Comment:
This is mutating the passed in argument which is not good. If you want to write to `options` you need to `dup` it first.